knowledge-base/grid-convert-descriptors-to-sql.md

```diff
@@ -0,0 +1,207 @@
+---
+title: Converting Filter and Sort Descriptors to SQL Queries in Blazor Grid
+description: Learn how to parse filter and sort descriptors from the Blazor Grid into SQL query statements for manual SQL queries execution using the OnRead event.
+type: how-to
+page_title: How to Parse Blazor Grid Descriptors into SQL Query Statements
+slug: grid-convert-descriptors-to-sql
+tags: grid, blazor, filter descriptors , sort descriptors, SQL query
+res_type: kb
+ticketid: 1666625, 1653361
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+When using the Grid [`OnRead` event]({%slug grid-events%}#read-event) to execute SQL queries, I need to convert the Grid's filter and sort descriptors into SQL query statements. This way I can create SQL clauses for filtering and ordering items directly through SQL. 
+
+This KB article also answers the following questions:
+- How can I convert Grid filters and sorters to SQL WHERE and ORDER BY clauses?
+- Is there a way to parse Grid filter and sort descriptors into SQL queries?
+- Can I use [`DataSourceRequest`]({%slug common-features-data-binding-onread%}#event-argument) to generate SQL query statements for filtering and sorting?
+
+## Solution
+
+To convert the Grid's filter and sort descriptors into SQL query statements, you need to manually construct the SQL query within the `OnRead` event handler by utilizing the `args.Request.Filters` and `args.Request.Sorts` objects. Although Telerik UI for Blazor does not provide a direct method to extract the SQL query from the `DataSourceRequest`, a manual approach can be adopted. 
+
+The following steps outline how to achieve this:
+
+1. **Handle the `OnRead` Event**: Add an `OnRead` event to your Grid and in the event handler, access the `args.Request.Filters` and `args.Request.Sorts` to construct your SQL query.
```

Comment:
The way the entire list is formed strays a bit from the way we tend to present procedures in our documentation. We usually go with an active, imperative sentence for the entire step. So for example, I would rework all steps to something like this (basically, drop the parts enclosed in bold formatting):

```suggestion
1. Add an `OnRead` event to your Grid and in the event handler, access the `args.Request.Filters` and `args.Request.Sorts` to construct your SQL query.
```

If you are keen on keeping them, remove the bold formatting and the title-style casing, and switch the colon to a period. 

```
1. Handle the `OnRead` event.<br/>Add an `OnRead` event to your Grid and in the event handler, access the `args.Request.Filters` and `args.Request.Sorts` to construct your SQL query.
```

Apply your preferred approach throughout the list.